test/unit/rubygem_test.rb

```diff
@@ -498,6 +500,7 @@ class RubygemTest < ActiveSupport::TestCase
         assert_equal @rubygem.linkset.mail, hash["mailing_list_uri"]
         assert_equal @rubygem.linkset.code, hash["source_code_uri"]
         assert_equal @rubygem.linkset.bugs, hash["bug_tracker_uri"]
+        assert_equal @rubygem.linkset.funding, hash["funding_uri"]
```

Comment:
so, linkset model is legacy and used in the fallback of Links (based on metadata). Just remove this change.

Perhaps you can add a unit test of the funding method and/or integration test for the link being visible when metadata is set.